lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -537,23 +525,46 @@ private void indexObject(
                     typeStorage.listObjectTypeParsingRules(storageObjectType));
             Collections.sort(parsingRules, new ParsingRulesSubtypeFirstComparator());
             for (final ObjectTypeParsingRules rule : parsingRules) {
-                final long t2 = System.currentTimeMillis();
-                final ParseObjectsRet parsedRet = parseObjects(guid, indexLookup,
-                        newRefPath, obj, rule);
-                long parsingTime = System.currentTimeMillis() - t2;
-                logger.logInfo(String.format("[Indexer]   Parsed %s %s in %s ms.",
-                        parsedRet.guidToObj.size(), toVerRep(rule.getGlobalObjectType()),
-                        parsingTime));
-                long t3 = System.currentTimeMillis();
-                indexObjectInStorage(guid, timestamp, isPublic, obj, rule,
-                        parsedRet.guidToObj, parsedRet.parentJson);
-                long indexTime = System.currentTimeMillis() - t3;
-                logger.logInfo("[Indexer]   " + toVerRep(rule.getGlobalObjectType()) +
-                        ", indexing time: " + indexTime + " ms.");
-                logger.timeStat(guid, 0, parsingTime, indexTime);
+
+                final SearchObjectType globalIdx = rule.getGlobalObjectType();
+
+                //pass into indexingStorage
+                boolean indexed = indexingStorage.hasParentId(globalIdx, guid);
+
+                if (indexed && !overwriteExistingData) {
+                    logger.logInfo("[Indexer]   skipping " + guid +
+                            " creation (already indexed and overwriteExistingData flag is set to false)");
```

Comment:
this message should include the search object type since it could be true for one SOT and not another SOT